### PR TITLE
[DINOv2] Add backbone class

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR:

- adds `Dinov2Backbone` as it was requested at https://github.com/NielsRogge/Transformers-Tutorials/issues/343 and https://github.com/facebookresearch/dinov2/issues/153#issuecomment-1678465000.
- adds DINOv2 to the documentation tests.

### CODE REVIEW

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding! 

Overall looks good - just a few pieces to tidy up e.g. old files and print statements

**Comment by NielsRogge** [comment]
@amyeroberts I've addressed your comments, feel free to approve/merge :)

File: src/transformers/models/dinov2/modeling_dinov2.py [hunk 1]
```diff
@@ -762,3 +763,103 @@ def forward(
...
+        self.post_init()
+
+    def get_input_embeddings(self) -> Dinov2PatchEmbeddings:
+        return self.embeddings.patch_embeddings
+
+    @add_start_docstrings_to_model_forward(DINOV2_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=BackboneOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        pixel_values: Tensor,
```
Comment (amyeroberts) [inline src/transformers/models/dinov2/modeling_dinov2.py:795]: nit: to match the other type hints in the module file 

```suggestion
        pixel_values: torch.Tensor,
```

**Review by amyeroberts — APPROVED** [review]
Thanks for adding! 

Just a tiny tiny nit re an unneeded import. Otherwise - good to merge!

**Pull request merged**